### PR TITLE
Prevent robot_state_publisher from publishing on a timer

### PR DESCRIPTION
**Pull request by wep21**
Backport a part of #158 to `galactic`
Refer https://github.com/ros/robot_state_publisher/pull/158#issuecomment-925802883

### CODE REVIEW

**Comment by wep21** [comment]
@clalancette Is the behavior change of this PR still too big to merge into `galactic`?

File: include/robot_state_publisher/robot_state_publisher.hpp [hunk 1]
```diff
@@ -95,7 +95,6 @@ class RobotStatePublisher : public rclcpp::Node
   rclcpp::Publisher<std_msgs::msg::String>::SharedPtr description_pub_;
   std::chrono::milliseconds publish_interval_ms_;
   rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_state_sub_;
-  rclcpp::TimerBase::SharedPtr timer_;
```
Comment (sloretz) [inline include/robot_state_publisher/robot_state_publisher.hpp]: Removing a class member breaks ABI, so this line can't be backported.

File: src/robot_state_publisher.cpp [hunk 2]
```diff
@@ -148,9 +148,7 @@ RobotStatePublisher::RobotStatePublisher(const rclcpp::NodeOptions & options)
       &RobotStatePublisher::callbackJointState, this,
       std::placeholders::_1));
 
-  // trigger to publish fixed joints
-  timer_ = this->create_wall_timer(
-    publish_interval_ms_, std::bind(&RobotStatePublisher::publishFixedTransforms, this));
```
Comment (sloretz) [inline src/robot_state_publisher.cpp:153]: IIUC if `use_tf_static` is False then the fixed transforms must be published regularly or they will not be received by late joining subscribers.

I think creating the timer only when `use_tf_static` is False would be a backportable change.

**Review by sloretz — CHANGES_REQUESTED** [review]

File: include/robot_state_publisher/robot_state_publisher.hpp (hunk 1 above)
Comment (wep21) [reply include/robot_state_publisher/robot_state_publisher.hpp]: @sloretz Fixed at https://github.com/ros/robot_state_publisher/pull/175/commits/091935468e0792b79b846d6fe401a2ef19275b56

File: src/robot_state_publisher.cpp (hunk 2 above)
Comment (wep21) [reply src/robot_state_publisher.cpp:153]: @sloretz Fixed at https://github.com/ros/robot_state_publisher/pull/175/commits/091935468e0792b79b846d6fe401a2ef19275b56

**Comment by wep21** [comment]
@sloretz @clalancette friendly ping

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15346/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10057/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13026/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15550/)

**Comment by clalancette** [comment]
Failures on Windows CI are unrelated to this change.  But since Windows is the most likely to have problems, I'm going to wait until Windows is fixed and re-run it before merging.

**Comment by wep21** [comment]
@clalancette Could you do bloom-release for `galactic`?

**Comment by clalancette** [comment]
Done in https://github.com/ros/rosdistro/pull/31336

**Pull request merged**